### DISCUSSION TITLE
How can I disable compound expressions in intellisense?

### DISCUSSION THREAD

**Discussion by beauxq** [microsoft/pylance-release#4783] [Q&A]
I want the auto-complete suggestions to not include any compound expressions. I want it to only include individual symbols.
![compound-expression-auto-complete](https://github.com/microsoft/pylance-release/assets/6052147/21c6e4ee-245b-419b-9965-4ef846d781f4)

I don't want it to include:
```
cursor + 2
cursor + 3:cursor + 5
cursor:cursor + 2
cursor:cursor + 3
cursor:cursor + size
```
`cursor` and everything below it in the screenshot are ok

**Comment by debonte** [comment]
For what it's worth, I believe it's including those options because you used those expressions to index into the same variable (`bytes`) somewhere above this line in your code.

Do you really want them to be excluded from the completions list (there are no scenarios where you would want these)? Or is this a prioritization issue? For example, is the problem that you would prefer local vars (ex. `cursor`) to appear above these expressions?

**Reply by beauxq** [reply]
I don't see myself ever wanting to use expressions like that at all.

**Reply by beauxq** [reply]
In addition to not wanting to use them, I also worry about the CPU costs and memory costs of them being there.
